runtime/metricsview/executor_timestamps.go

```diff
@@ -0,0 +1,243 @@
+package metricsview
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"time"
+
+	"github.com/rilldata/rill/runtime/drivers"
+	"golang.org/x/sync/errgroup"
+)
+
+const (
+	defaultExecutionTimeout = time.Minute * 3
+)
+
+func (e *Executor) resolveDuckDB(ctx context.Context) (time.Time, time.Time, time.Time, error) {
+	timeDim := e.olap.Dialect().EscapeIdentifier(e.metricsView.TimeDimension)
+	escapedTableName := e.olap.Dialect().EscapeTable(e.metricsView.Database, e.metricsView.DatabaseSchema, e.metricsView.Table)
+
+	var watermarkExpr string
+	if e.metricsView.WatermarkExpression != "" {
+		watermarkExpr = e.metricsView.WatermarkExpression
+	} else {
+		watermarkExpr = fmt.Sprintf("MAX(%s)", timeDim)
+	}
+
+	rangeSQL := fmt.Sprintf(
+		"SELECT min(%[1]s) as \"min\", max(%[1]s) as \"max\", %[2]s as \"watermark\", max(%[1]s) - min(%[1]s) as \"interval\" FROM %[3]s %[4]s",
```

Comment:
`interval` is not used in the result here. If you remove it, I think the implementation here can be combined with `resolveClickHouseAndPinot`